spec/models/search_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "rails_helper"
+
+RSpec.describe Search do
+  describe "#search_laa_reference" do
+    let(:search) { described_class.search(query).pluck(:id) }
+    let(:prepare) { build(:submission).tap(&:save) }
+
+    before { prepare }
+
+    context "when search text is LAA reference" do
+      let(:query) { "LAA-123456" }
+
+      it "returns the record" do
+        expect(search).to eq([prepare.id])
+      end
+    end
+
+    context "when search text is LAA reference does not match any records" do
+      let(:query) { "LAA-121212" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is full number part of LAA reference" do
+      let(:query) { "123456" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is partial number part of LAA reference" do
+      let(:query) { "345" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is 'laa'" do
+      let(:query) { "laa" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
```

Comment:
should we add a test for lass reference with name..

i.e. let(:query) { "LAA-123456 joe bloggs" }

@jogilder we have made some assumptions on how matching will work that we likely need to check (or at least document)